cvat/apps/engine/media_extractors.py

```diff
@@ -868,16 +876,17 @@ def _compress_image(source_image: av.VideoFrame | io.IOBase | Image.Image, quali
             image = Image.fromarray(image, mode="L") # 'L' := Unsigned Integer 8, Grayscale
             image = ImageOps.equalize(image)         # The Images need equalization. High resolution with 16-bit but only small range that actually contains information
 
-        converted_image = image.convert('RGB')
+        if image.mode != 'RGB' and image.mode != 'L':
+            image = image.convert('RGB')
 
         try:
             buf = io.BytesIO()
-            converted_image.save(buf, format='JPEG', quality=quality, optimize=True)
+            image.save(buf, format='JPEG', quality=quality, optimize=True)
             buf.seek(0)
-            width, height = converted_image.size
-            return width, height, buf
+            return image.width, image.height, buf
         finally:
-            converted_image.close()
+            if image is not source_image:
+                image.close()
```

Comment:
Is this `close` even needed at all anymore? As far as I can see, it's impossible for `image` to have an attached open file at this point (given the changes to `load_image`).